presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java

```diff
@@ -81,6 +82,9 @@
     @GuardedBy("this")
     private long peakNodeTotalMemory;
 
+    @GuardedBy("this")
+    private long maxRevocableMemory;
```

Comment:
Make it final and put it with `private final` constants. Maybe add a TODO to say this will be tunable from session properties in the future.